src/main/java/org/spburegistry/backend/service/implementation/ClientServiceImpl.java

```diff
@@ -0,0 +1,68 @@
+package org.spburegistry.backend.service.implementation;
+
+import java.util.stream.Collectors;
+
+import org.spburegistry.backend.dto.ClientTO;
+import org.spburegistry.backend.dto.UserTO;
+import org.spburegistry.backend.entity.Client;
+import org.spburegistry.backend.entity.User;
+import org.spburegistry.backend.enums.Role;
+import org.spburegistry.backend.repository.ClientRepo;
+import org.spburegistry.backend.repository.UserRepo;
+import org.spburegistry.backend.service.ClientService;
+import org.spburegistry.backend.utils.ConvertToTO;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import jakarta.persistence.EntityNotFoundException;
+
+@Service
+public class ClientServiceImpl implements ClientService {
+
+    private final ClientRepo clientRepo;
+    private final UserRepo userRepo;
+
+    @Autowired
+    public ClientServiceImpl(ClientRepo clientRepo, UserRepo userRepo) {
+        this.clientRepo = clientRepo;
+        this.userRepo = userRepo;
+    }
+
+    @Override
+    public Iterable<UserTO> findAll() {
+        return userRepo.findAll().stream()
+                .filter(user -> user.getStudent() == null)
+                .map(ConvertToTO::userToTO)
+                .collect(Collectors.toSet());
+    }
+
+    @Override
+    public UserTO findById(long id) {
+        Client client = clientRepo.findById(id)
+                .orElseThrow(() -> new EntityNotFoundException("Client with id " + id + " not found"));
+        return ConvertToTO.userToTO(client.getUser());
```

Comment:
```suggestion
        return clientRepo.findById(id)
                .map(Client::getUser)
                .map(ConvertToTO::userToTO)
                .orElseThrow(() -> new EntityNotFoundException("Client with id " + id + " not found"));
```
and remove 40 and 41 lines